File: LeetCode/Majority Element.cpp
class Solution {
public:
    int majorityElement(vector<int> &num) {
		int n = num.size();
		int cnt = 1;
		int ret = num[0];

		for (int i = 1; i < n; i++) {
			if (num[i] == ret) {
				++cnt;
			} else {
				--cnt;
				if (!cnt) {
					cnt = 1;
					ret = num[i];
				}
			}
		}

		return ret;
    }
};

File: LeetCode_Java/src/MajorityElement.java
public class MajorityElement {
    public int majorityElement(int[] nums) {
        int element = nums[0];
        int count = 0;
        for (int num : nums) {
            if (count == 0) {
                element = num;
                count = 1;
            } else {
                if (element != num) {
                    count--;
                } else {
                    count++;
                }
            }
        }

        return element;
    }
}
